src/proto/single/ent.rs

```diff
@@ -1,6 +1,23 @@
 use chrono::{DateTime, Utc};
+use derive_builder::Builder;
+use serde::{
+    de::{Deserializer, IntoDeserializer},
+    Deserialize,
+};
 
-use crate::JobBuilder;
+use crate::{Error, JobBuilder};
+
+// Used to parse responses from Faktory that look like this:
+// '{"jid":"f7APFzrS2RZi9eaA","state":"unknown","updated_at":""}'
```

Comment:
So am I right in assuming that the reason you need this helper is really just _specifically_ to handle the case where a datetime field is _set_  but _empty_? If so, let's be more explicit here. It wasn't clear to me reading this that you hadn't pasted the wrong thing and intended to paste a timestamp example :sweat_smile:

Reply:
Correct, specifically for this case: empty string :)